### PR TITLE
Fixing hard-coded references to driver versions, misc. backports.

### PR DESCRIPTION
**Pull request by ianjevans**
This PR removes hard-coded version numbers from the driver/framework docs. There are some additional filter fixes and backports, and updates for psycopg3.

Fixes DOC-4692.

### CODE REVIEW

**Review by dbist — COMMENTED** [review]
> liquibase-4.2.0-bin/lib/

should probably also say <version> as long as 4.2 is minimum.

**Review by nickvigilante — APPROVED** [review]
Approved.

Also @nickvigilante check all existing references to "ActiveRecord" and prefer "Active Record" in Vale.

**Comment by ianjevans** [comment]
> > liquibase-4.2.0-bin/lib/
> 
> should probably also say as long as 4.2 is minimum.

Earlier in the topic we say 4.2.0 or greater.

File: v22.1/liquibase.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Before you begin the tutorial, do the following:
 
 To install the Liquibase binary on your machine:
 
-1. Download the latest version of the [Liquibase command-line tool](https://www.liquibase.org/download). CockroachDB is fully compatible with Liquibase versions 4.2.0 and greater. We use the binary download of Liquibase 4.20, for macOS.
+1. Download the latest version of the [Liquibase command-line tool](https://www.liquibase.org/download). CockroachDB is fully compatible with Liquibase versions 4.2.0 and greater. This tutorial uses the binary download of Liquibase 4.2.0, for macOS.
```
Comment (gemma-shay) [inline v22.1/liquibase.md:23]:
```suggestion
1. Download the latest version of the [Liquibase command-line tool](https://www.liquibase.org/download). CockroachDB is fully compatible with Liquibase versions 4.2.0 and greater. This tutorial uses the binary download of Liquibase 4.2.0 for macOS.
```

File: v22.2/liquibase.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ Before you begin the tutorial, do the following:
 
 To install the Liquibase binary on your machine:
 
-1. Download the latest version of the [Liquibase command-line tool](https://www.liquibase.org/download). CockroachDB is fully compatible with Liquibase versions 4.2.0 and greater. We use the binary download of Liquibase 4.20, for macOS.
+1. Download the latest version of the [Liquibase command-line tool](https://www.liquibase.org/download). CockroachDB is fully compatible with Liquibase versions 4.2.0 and greater. This tutorial uses the binary download of Liquibase 4.2.0, for macOS.
```
Comment (gemma-shay) [inline v22.2/liquibase.md:23]:
```suggestion
1. Download the latest version of the [Liquibase command-line tool](https://www.liquibase.org/download). CockroachDB is fully compatible with Liquibase versions 4.2.0 and greater. This tutorial uses the binary download of Liquibase 4.2.0 for macOS.
```

**Review by gemma-shay — APPROVED** [review]
just one nit but LGTM!

**Pull request merged**